### PR TITLE
Bug 574363 - Move jdt.junit* bundles to Java 11

### PR DESCRIPTION
**Pull request by iloveeclipse**
Originally created as
https://git.eclipse.org/r/c/jdt/eclipse.jdt.ui/+/182392

### CODE REVIEW

**Comment by iloveeclipse** [comment]
???
```
SNAPSHOT:compile (default-compile) on project org.eclipse.jdt.junit.runtime: Compilation failure: Compilation failure: 
11:48:22  [ERROR] /home/jenkins/agent/workspace/eclipse.jdt.ui-github_PR-3/org.eclipse.jdt.junit.runtime/src/org/eclipse/jdt/internal/junit/runner/ITestReference.java: 
11:48:22  [ERROR] 	/*******************************************************************************
11:48:22  [ERROR] 	^
11:48:22  [ERROR] Type java.lang.Object is indirectly referenced from required .class files but cannot be resolved since the declaring package java.lang exported from module java.base conflicts with a package accessible from module <unnamed>
11:48:22  [ERROR] 1 problem (1 error)
```

File: org.eclipse.jdt.junit5.runtime/build.properties [hunk 1]
```diff
@@ -18,3 +18,4 @@ bin.includes = about.html,\
                plugin.properties,\
                .,\
                META-INF/
+jre.compilation.profile = JavaSE-1.8
```
Comment (vogella) [inline org.eclipse.jdt.junit5.runtime/build.properties:21]: Why change BREE but add a 1.8 compilation profile? If that is intended, please document via commit message.

**Comment by ktatavarthi** [comment]
@iloveeclipse I donot have much idea regarding `jre.compilation.profile = JavaSE-1.8` , but as there is no pom change and bree changes to Java 11, will Maven-Tycho build use the above attribute to use --release option?

File: org.eclipse.jdt.junit5.runtime/build.properties (hunk 1 above)
Comment (iloveeclipse) [reply org.eclipse.jdt.junit5.runtime/build.properties:21]: > Why change BREE but add a 1.8 compilation profile? If that is intended, please document via commit message.

See https://bugs.eclipse.org/bugs/show_bug.cgi?id=574363

Comment (vogella) [reply org.eclipse.jdt.junit5.runtime/build.properties:21]: Please document via the commit message.

**Comment by ktatavarthi** [comment]
> > will Maven-Tycho build use the above attribute to use --release option?
> 
> I have no idea what Tycho/Maven will do. If no one knows if compiling on java 11 with target 8 is possible, we can either drop this idea entirely and keep the bundle on 8 or move to 11 but that would mean 8 will be not supported for test.

I have experimented locally and the below code added to the pom files of the three junit runtime plugins makes sure that the tycho maven build uses the --release option 

```
  <build>
	<plugins>
		<plugin>
		 	<groupId>org.eclipse.tycho</groupId>
			<artifactId>tycho-compiler-plugin</artifactId>
			<configuration>
				<release>8</release>
			</configuration>
		</plugin>
	</plugins>
  </build>
```

**Comment by ktatavarthi** [comment]
> > > will Maven-Tycho build use the above attribute to use --release option?
> > 
> > 
> > I have no idea what Tycho/Maven will do. If no one knows if compiling on java 11 with target 8 is possible, we can either drop this idea entirely and keep the bundle on 8 or move to 11 but that would mean 8 will be not supported for test.
> 
> I have experimented locally and the below code added to the pom files of the three junit runtime plugins makes sure that the tycho maven build uses the --release option
> 
> ```
>   <build>
> 	<plugins>
> 		<plugin>
> 		 	<groupId>org.eclipse.tycho</groupId>
> 			<artifactId>tycho-compiler-plugin</artifactId>
> 			<configuration>
> 				<release>8</release>
> 			</configuration>
> 		</plugin>
> 	</plugins>
>   </build>
> ```

The above change would be in addition to all the changes that are already done in this PR.

**Comment by iloveeclipse** [comment]
> The above change would be in addition to all the changes that are already done in this PR.

Cool, thanks, I've rebased now.

**Review by jjohnstn — APPROVED** [review]
LGTM

**Comment by iloveeclipse** [comment]
Thanks Jeff.

**Pull request merged**